### PR TITLE
Remove unused CComVariant from InstrumentationEngine.Lib

### PR DESCRIPTION
**Pull request by schmittjoseph**
- Remove an unused `CComVariant` variable that was resulting in `ClrInstrumentationEngine.so` exporting `clrie_VariantInit`.
- Remove `CComVariant` related code from unix headers so that future instances of this issue should result in a build break.
- Fix typo relating to the Linux build script.

### CODE REVIEW

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:

File: src/unix/inc/atl/atlcomcli.h [hunk 1]
```diff
@@ -1470,1495 +1470,1495 @@ class CAdapt
 // CComVariant
```
Comment (WilliamXieMSFT) [inline src/unix/inc/atl/atlcomcli.h:1470]: >CComVariant [](http://example.com/codeflow?start=3&length=11)

maybe add a comment about not implemented in PAL

File: src/unix/inc/atl/atlcoll.h [hunk 2]
```diff
@@ -238,42 +238,42 @@ class CElementTraits< GUID > :
...
-			return( 1 );
-		}
-		else
-		{
-			ATLASSERT( hr == static_cast<HRESULT>(VARCMP_EQ) || hr == static_cast<HRESULT>(VARCMP_NULL) );
-			return( 0 );
-		}
-	}
-};
+// template<>
```
Comment (delmyers) [inline src/unix/inc/atl/atlcoll.h:241]: >//  [](http://example.com/codeflow?start=0&length=3)

Please add a comment indicating that the PAL does not implement CComVariant, so we do not allow the type to be referenced on non-windows platforms.

**Review by delmyers — APPROVED** [review]
:shipit:

**Pull request merged**